File: src/main/java/com/test/task/services/interfaces/DistrictService.java
package com.test.task.services.interfaces;

import com.test.task.model.District;
import org.springframework.http.ResponseEntity;

public interface DistrictService {
    Iterable<District> getDistricts();
    ResponseEntity<District> addDistrict(District district);
    void deleteDistrict(Long id);
    ResponseEntity<District> updateDistrict(Long id, District district);
}

File: test_task_front/src/app/services/district.service.ts
import {Injectable} from "@angular/core";
import {HttpClient} from "@angular/common/http";
import {District} from "../model/district";
import {Observable} from "rxjs";

@Injectable()
export class DistrictService {

  constructor(private http: HttpClient) {
  }

  private url = "http://localhost:8080/district";

  getDistricts(): Observable<District[]> {
    return this.http.get<District[]>(`${this.url}/get`);
  }

  addDistrict(district: District): Observable<District> {
    return this.http.post<District>(`${this.url}/add`, district);
  }

  deleteDistrict(id: number): Observable<void> {
    return this.http.delete<void>(`${this.url}/delete/${id}`);
  }

  updateDistrict(id: number, district: District): Observable<District> {
    return this.http.put<District>(`${this.url}/update/${id}`, district);
  }
}
